server/app/controllers/receive.py

```diff
@@ -0,0 +1,118 @@
+import abc
+import datetime
+import os.path
+import uuid
+
+from flask import request, abort
+from webargs import fields
+from webargs.flaskparser import use_args
+from werkzeug.utils import secure_filename
+
+from app import app
+from app.authorize_station import authorize_station
+from app.utils import make_thumbnail, first
+from app.repository import Observation, ObservationFile, ObservationFileId, ObservationId, Repository, SatelliteId, StationId
+from abc import abstractmethod
+
+import sys
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+class WebFileLike(abc.ABC):
+    @property
+    @abstractmethod
+    def filename(self) -> str:
+        pass
+    @abstractmethod
+    def save(self, path: str) -> None:
+        pass
+
+class RequestArguments(TypedDict):
+    aos: datetime.datetime
+    tca: datetime.datetime
+    los: datetime.datetime
+    sat: str
+    notes: str
+
+ALLOWED_FILE_TYPES = {
+    "image/png": ".png"
+}
+
+get_extension = lambda p: os.path.splitext(p)[1]
+is_allowed_file = lambda f: f.mimetype in ALLOWED_FILE_TYPES and \
+                            ALLOWED_FILE_TYPES[f.mimetype] == get_extension(f.filename)
+
+@app.route('/receive', methods=["POST",])
+@authorize_station
+@use_args({
+    'aos': fields.DateTime(required=True),
+    'tca': fields.DateTime(required=True),
+    'los': fields.DateTime(required=True),
+    'sat': fields.Str(required=True),
+    'notes': fields.Str(required=False)
+})
+def receive(station_id: str, args: RequestArguments):
+    if len(request.files) == 0:
```

Comment:
This is a complicated method. It should have a description that explains what exactly is being done.

Reply:
I added the description.